app/modules.go

```diff
@@ -128,3 +133,57 @@ func (SlashingModule) DefaultGenesis(cdc codec.JSONCodec) json.RawMessage {
 
 	return cdc.MustMarshalJSON(genState)
 }
+
+// GenutilModule defines a custom wrapper around the x/genutil module's
+// AppModuleBasic implementation to provide custom genesis state validation.
+type GenutilModule struct {
+	genutil.AppModuleBasic
+}
+
+// ValidateGenesis validates the x/genutil genesis state.
+func (GenutilModule) ValidateGenesis(
+	cdc codec.JSONCodec,
+	encCfg client.TxEncodingConfig,
+	bz json.RawMessage,
+) error {
+
+	var genState genutiltypes.GenesisState
+	if err := cdc.UnmarshalJSON(bz, &genState); err != nil {
+		return fmt.Errorf("failed to unmarshal %s genesis state: %w", genutiltypes.ModuleName, err)
+	}
+
+	txJSONDecoder := encCfg.TxJSONDecoder()
+
+	for i, genTx := range genState.GenTxs {
+		var tx sdk.Tx
+
+		tx, err := txJSONDecoder(genTx)
+		if err != nil {
+			return err
+		}
+
+		msgs := tx.GetMsgs()
+		if n := len(msgs); n != 2 {
+			return fmt.Errorf(
+				"gentx %d contains invalid number of messages; expected: 2; got: %d",
+				i, n,
+			)
+		}
+
+		if _, ok := msgs[0].(*stakingtypes.MsgCreateValidator); !ok {
+			return fmt.Errorf(
+				"gentx %d contains invalid message at index 0; expected: %T; got: %T",
+				i, &stakingtypes.MsgCreateValidator{}, msgs[0],
+			)
+		}
+
+		if _, ok := msgs[1].(*gravitytypes.MsgSetOrchestratorAddress); !ok {
+			return fmt.Errorf(
+				"gentx %d contains invalid message at index 1; expected: %T; got: %T",
+				i, &gravitytypes.MsgSetOrchestratorAddress{}, msgs[0],
```

Comment:
```suggestion
				i, &gravitytypes.MsgSetOrchestratorAddress{}, msgs[1],
```
Is this error message correct?

Reply:
good catch!